internal/controllers/provisioningrequest_controller.go

```diff
@@ -2558,3 +2520,46 @@ func (r *ProvisioningRequestReconciler) SetupWithManager(mgr ctrl.Manager) error
 			})).
 		Complete(r)
 }
+
+const (
+	// Policy Specific states
+	compliant    = "Compliant"
+	nonCompliant = "NonCompliant"
+	pending      = "Pending"
+)
+
+// BuildPoliciesEventHistory creates an EventHistory object from a policyList
+func BuildPoliciesEventHistory(policies *policiesv1.PolicyList) (h utils.EventHistory) {
+	for _, policy := range policies.Items {
+		for _, detailsPerTemplate := range policy.Status.Details {
```

Comment:
Same as above, if the current detail is Compliant, it should be skipped.

Reply:
how would we accurately calculate the timeout in the case describe in the diagram? same as https://github.com/openshift-kni/oran-o2ims/pull/246#discussion_r1803865576